Massive documentation overhaul

Quaker762 opened the pull request:
Fixed all .hpp file comments to conform to the Doxygen standard.

Quaker762 committed:
Massive documentation overhaul

Fixed all .hpp file comments to conform to the Doxygen standard.

z33ky commented on include/SH3/arc/section.hpp:
```diff
@@ -1,45 +1,45 @@
-#define ARCSECTION_MAGIC 0x20030507
+#define ARCSECTION_MAGIC 0x20030507 /**< Magic number (first 4 bytes) of an .arc section */
 
+/** @defgroup arc-headers arc header types
```
sub-arc headers*

z33ky commented:
> all .hpp

util.hpp is missing (an is empty besides comments, so should probably just be removed).
types/vertex.hpp (currently unused, but has code) and system/sdl_destroyer.hpp are missing.

Quaker762 committed:
Axed 'util.hpp'

There's nothing in there, so it gets the axe.

Quaker762 committed:
Added documentation to 'vertex.hpp'

Forgot this one... Oops....

Quaker762 committed:
Added documentation to 'sdl_destroyer.hpp'

Forgot this one as well.....

Quaker762 committed:
Fixed a typo in 'section.hpp'

Put all sub arc types into a doxygroup called 'arc-section-headers'
instead of arc-headers.
